pyresample/geometry.py

```diff
@@ -475,6 +475,57 @@ def __str__(self):
                                                     str(self.lons),
                                                     str(self.lats))
 
+    def geocentric_resolution(self, ellps='WGS84', radius=None, nadir_factor=2):
+        """Calculate maximum geocentric pixel resolution.
+
+        If `lons` is a :class:`xarray.DataArray` object with a `resolution`
+        attribute, this will be used instead of loading the longitude and
+        latitude data. In this case the resolution attribute is assumed to
+        mean the nadir resolution of a swath and will be multiplied by the
+        `nadir_factor` to adjust for increases in the spatial resolution
+        towards the limb of the swath.
+
+        Args:
+            ellps (str): PROJ Ellipsoid for the Cartographic projection
+                used as the target geocentric coordinate reference system.
+                Default: 'WGS84'. Ignored if `radius` is provided.
+            radius (float): Spherical radius of the Earth to use instead of
+                the definitions in `ellps`.
+            nadir_factor (int): Number to multiply the nadir resolution
+                attribute by to reflect pixel size on the limb of the swath.
+
+        Returns: Estimated maximum pixel size in meters on a geocentric
+            coordinate system (X, Y, Z) representing the Earth.
+
+        Raises: RuntimeError if a simple search for valid longitude/latitude
+            data points found no valid data points.
+
+        """
+        if hasattr(self.lons, 'attrs') and 'resolution' in self.lons.attrs:
+            return self.lons.attrs['resolution'] * nadir_factor
+
+        from pyproj import transform
+        rows, cols = self.shape
```

Comment:
This makes the assumption that the data is two dimensional, but it might not be the case.

Reply:
You're right. I wasn't thinking of the 1D case. Should I also expect/handle a 3D case?

Reply:
I don't really see how a lon array could be 3D...

Reply:
Sounding data (IASI, CRIS) could (should) have 3D coordinates. Unfortunately they don't at the moment.

Reply:
I'm still worried this might break eg lightning data resampling

Reply:
so in the != 2d case, this should also raise a Runtime error I guess

Reply:
Yep, I forgot to read these comments again before saying it was ready for review. You're right that the 1D case is difficult. In the lightning case we don't want to calculate a resolution because each point is at a "random" location. For NUCAPS though it is a 1D array of swath pixels so we can make some assumptions. That said, I think you're right that a Runtime Error may be the safest here.

Reply:
This should be fixed now.